File: Python/quick_sort.py
import time
import random

def partition(lista, left, right):
    pivot = lista[right]
    i = left-1
    for j in range(left, right):
        if lista[j] <= pivot:
            i += 1
            lista[i], lista[j] = lista[j], lista[i]
    i += 1
    lista[i], lista[right] = lista[right], lista[i]
    return i


def quick_sort(lista, left, right):
    if left < right:
        pivot = partition(lista, left, right)
        quick_sort(lista, left, pivot-1)
        quick_sort(lista, pivot+1, right)

if __name__ == '__main__':
	lista = [_ for _ in range(1000)]
	random.shuffle(lista)
	start_time = time.time()
	quick_sort(lista, 0, len(lista) - 1)
	end_time = time.time()
	print("Time: ", end_time - start_time)
	
	

File: javaSorting/QuickSort.java
package javaSorting;

public class QuickSort {

	public static int partition(int data[], int left, int right) {
		int pivot = data[right];
		int i = left - 1;
		for (int j = left; j < right; j++) {
			if (data[j] <= pivot) {
				i++;
				int temp = data[i];
				data[i] = data[j];
				data[j] = temp;
			}
		}
		i++;
		int temp = data[i];
		data[i] = data[right];
		data[right] = temp;
		return i;
	}

	public static void sort(int data[], int left, int right) {
		if (left < right) {
			int pivot = partition(data, left, right);
			sort(data, left, pivot - 1);
			sort(data, pivot + 1, right);
		}
	}

	private static void printSortedArray(int data[]) {
		for (int i = 0; i < data.length; i++) {
			System.out.println(data[i]);
		}
	}

	public static void main(String[] args) {
		int SIZE = 1000;
		int[] data = new int[SIZE];
		for (int i = 0; i < SIZE; i++) {
			data[i] = i;
		}
		Shuffle.shuffleData(data);

		long start = System.currentTimeMillis();
		sort(data, 0, SIZE - 1);
		float elapsed = (System.currentTimeMillis() - start) / 1000;

		printSortedArray(data);
		System.out.println(elapsed);
	}
}
